knowledge/pkg/vectorstore/sqlite-vec/sqlite-vec.go

```diff
@@ -43,23 +47,18 @@ func New(ctx context.Context, dsn string, embeddingFunc cg.EmbeddingFunc) (*Vect
 		embeddingsTableName: "knowledge_embeddings",
 	}
 
-	stmt, _, err := db.Prepare(`SELECT sqlite_version(), vec_version()`)
-	if err != nil {
-		return nil, fmt.Errorf("failed to initialize sqlite-vec: %w", err)
-	}
-
-	stmt.Step()
-	slog.Debug("sqlite-vec info", "sqlite_version", stmt.ColumnText(0), "vec_version", stmt.ColumnText(1))
-	err = stmt.Close()
-	if err != nil {
-		return nil, err
+	var sqliteVersion, vecVersion string
+	res := db.Raw("SELECT sqlite_version(), vec_version()").Row()
+	if err := res.Scan(&sqliteVersion, &vecVersion); err != nil {
+		return nil, fmt.Errorf("failed to scan results: %w", err)
 	}
+	slog.Debug("sqlite-vec info", "sqlite_version", sqliteVersion, "vec_version", vecVersion)
 
 	return store, store.prepareTables(ctx)
 }
 
 func (v *VectorStore) Close() error {
-	return v.db.Close()
+	return nil
```

Comment:
Not sure how important this is, but it is possible to get the underlying database connection from gorm and close it.

Reply:
I remembered a comment about it not being recommended: https://github.com/go-gorm/gorm/issues/3145#issuecomment-658230163
.. however, since we run knowledge as one-off commands, this is certainly different for us.
I figured I never called any Close() method anywhere, so I now added it all throughout the code :+1: